assets/src/data/helpers/site-data.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External imports
+ */
+import { data } from '@eventespresso/eejs';
+
+data.site_formats = data.site_formats || {};
+
+/**
+ * All available site formats exposed via the eejs.data global from the server.
+ * @type {{}}
+ */
+export const { date_formats: dateFormats = {} } = data.site_formats;
+
+/**
+ * The date format used by the site or mysql date format if not set.
+ * @type { string }
+ */
+export const FORMAT_SITE_DATE = dateFormats.moment_split &&
+	dateFormats.moment_split.date ?
+	dateFormats.moment_split.date :
+	'YY-MM-DD';
```

Comment:
> I'm imagining a scenario where the site format is not getting set on a particular page for whatever reason

This won't happen because of the dependency set for the helpers (at least in a WP context).
The purpose of this is to have a human readable format.  Currently, the moment.default is ISO8601.  I figured that wasn't a really human friendly format and mysql would be more neutral.  I'm certainly open to changing the default to something else though.